elastic/beats
# Ignore https errors

Fixes https://github.com/elastic/beats/issues/27202

NOTE: This requires a version of synthetics with https://github.com/elastic/synthetics/pull/361

Adds an option to disable errors on invalid TLS certificates in heartbeat.

Rather than try to work with `ssl.verification_mode: none` this adds a new option. The reason being that the behavior here is just too different to share a configuration option.


## Checklist


## How to test this PR locally

Badssl targets should function

```yaml
- id: test
  name: A Test
  type: browser
  schedule: "@every 10s"
  ignore_https_errors: true
  source:
    inline:
      script: |-
       step('Go to home page', async () => {
         await page.goto('https://expired.badssl.com/');
       });  
```
